Sources/PklSwift/Decoder/OptionalDictionaryKey.swift

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+public struct OptionalDictionaryKey<Wrapped>: Hashable, Decodable, CodingKeyRepresentable where Wrapped: Hashable, Wrapped: Decodable, Wrapped: CodingKeyRepresentable {
```

Comment:
This is a wrapper type for optional Dictionary key types (other than `AnyHashable`) that conforms to `CodingKeyRepresentable`, allowing them to be decoded correctly. This adds some indirection to generated code, but is necessary to have affected values decode correctly.

Dictionary has 4 decoding paths:
* Special case for `String` keys
* Special case for `Int` keys
* Key conforms to `CodingKeyRepresentable`
* Fallback: assume encoded value is an unkeyed collection storing [k1, v1, k2, v2]

The code generator will only use this type when it's required for correctness. In unaffected cases the key will not be wrapped.